frontend/public/components/monitoring/targets.tsx

```diff
@@ -0,0 +1,375 @@
+import * as _ from 'lodash-es';
+import { Alert } from '@patternfly/react-core';
+import { sortable } from '@patternfly/react-table';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { useTranslation } from 'react-i18next';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useSelector } from 'react-redux';
+import { Link, Route, RouteComponentProps, Switch, withRouter } from 'react-router-dom';
+
+import { PrometheusEndpoint } from '@console/dynamic-plugin-sdk/src/api/internal-types';
+import { GreenCheckCircleIcon, RedExclamationCircleIcon } from '@console/shared';
+
+import { NamespaceModel, ServiceModel, ServiceMonitorModel } from '../../models';
+import { K8sResourceKind, referenceForModel } from '../../module/k8s';
+import { RootState } from '../../redux';
+import { RowFunctionArgs, Table, TableData } from '../factory';
+import { FilterToolbar, RowFilter } from '../filter-toolbar';
+import { PROMETHEUS_BASE_PATH } from '../graphs';
+import { SectionHeading, BreadCrumbs } from '../utils/headings';
+import { useK8sWatchResource } from '../utils/k8s-watch-hook';
+import { usePoll } from '../utils/poll-hook';
+import { ResourceLink } from '../utils/resource-link';
+import { useSafeFetch } from '../utils/safe-fetch-hook';
+import { LoadingInline, StatusBox } from '../utils/status-box';
+import { Timestamp } from '../utils/timestamp';
+import { Labels } from './labels';
+import { AlertSource, PrometheusAPIError, Target } from './types';
+import { targetSource } from './utils';
+
+const MonitorsWatchContext = React.createContext([]);
+const ServicesWatchContext = React.createContext([]);
+
+const ServiceMonitor: React.FC<{ target: Target }> = ({ target }) => {
+  const [monitors, monitorsLoaded] = React.useContext(MonitorsWatchContext);
+  const [services, servicesLoaded] = React.useContext(ServicesWatchContext);
+
+  if (!servicesLoaded || !monitorsLoaded) {
+    return <LoadingInline />;
+  }
+
+  // First find the service that corresponds to the target
+  const service = _.find(
+    services,
+    ({ metadata }) =>
+      metadata.name === target?.labels?.service && metadata.namespace === target?.labels?.namespace,
+  );
+
+  // Now find the service monitor that corresponds to the service
+  const monitor = _.find(
+    monitors,
+    ({ spec }) =>
+      service &&
+      _.isMatch(service.metadata.labels, spec.selector?.matchLabels) &&
```

Comment:
I missed this previously, but what if the service uses `matchExpressions`?

I'd use `LabelSelector` from label-selector.js to handle `matchExpressions`.